community/kernel/src/test/java/org/neo4j/kernel/impl/api/state/TxStateTest.java

```diff
@@ -278,12 +284,57 @@ public void shouldAddAndGetByLabel()
     public void shouldAddAndGetByRuleId()
     {
         // GIVEN
-        state.indexRuleDoAdd( indexOn_1_1 );
+        state.indexRuleDoAdd( indexOn_1_1, null );
 
         // THEN
         assertEquals( asSet( indexOn_1_1 ), state.indexChanges().getAdded() );
     }
 
+    @Test
+    public void shouldRememberSpecificallySetIndexProviderDescriptor() throws ConstraintValidationException, CreateConstraintFailureException
```

Comment:
Please add test for create index with one provider, dropping it and creating it again with a different provider.